forge-ai/src/main/java/forge/ai/CreatureEvaluator.java

```diff
@@ -224,30 +226,66 @@ else if (c.hasKeyword(Keyword.WITHER)) {
         } else {
             value -= subValue(10 * c.getCounters(CounterEnumType.STUN), "stunned");
         }
-        if (c.hasSVar("EndOfTurnLeavePlay")) {
-            value -= subValue(50, "eot-leaves");
-        } else if (c.hasKeyword(Keyword.CUMULATIVE_UPKEEP)) {
-            value -= subValue(30, "cupkeep");
-        } else if (c.hasStartOfKeyword("UpkeepCost")) {
-            value -= subValue(20, "sac-unless");
-        } else if (c.hasKeyword(Keyword.ECHO) && c.cameUnderControlSinceLastUpkeep()) {
-            value -= subValue(10, "echo-unpaid");
-        }
-        if (c.hasKeyword(Keyword.FADING)) {
-            value -= subValue(20 / (Math.max(1, c.getCounters(CounterEnumType.FADE))), "fading");
-        }
-        if (c.hasKeyword(Keyword.VANISHING)) {
-            value -= subValue(20 / (Math.max(1, c.getCounters(CounterEnumType.TIME))), "vanishing");
-        }
         // use scaling because the creature is only available halfway
         if (c.hasKeyword(Keyword.PHASING)) {
             value -= subValue(Math.max(20, value / 2), "phasing");
         }
 
-        // TODO no longer a KW
-        if (c.hasStartOfKeyword("At the beginning of your upkeep, CARDNAME deals")) {
-            value -= subValue(20, "upkeep-dmg");
-        } 
+        if (c.hasSVar("EndOfTurnLeavePlay")) {
+            value -= subValue(50, "eot-leaves");
+        } else if (c.isInPlay()) {
```

Comment:
the rating should be worse even before it's in play otherwise AI won't consider it in spell decision